### DISCUSSION TITLE
Smooth font generation challenge

### DISCUSSION THREAD

**Discussion by u4mzu4** [Bodmer/TFT_eSPI#962] [Q&A - Fonts] [answered]
Hello!

I'm on a home project preparation phase, and would like to add my favorite Logisoso font to the SPIFFS as custom smooth font. After installed to the system, I tried the Create_font.pde in the Processing tool.
- If I add the font name manually, the script is running properly (fontNumber= -1)
but
- If I choose font from the the System_Font_List, the script failed, becuse the fontType is null here:
  
// Set the fontName from the array number or the defined fontName
  if (fontNumber >= 0)
  {
    fontName = fontList[fontNumber];
    fontType = "";
  }

After deleted the [line](https://github.com/Bodmer/TFT_eSPI/blob/b1fb969e86c328ab21dbb71c01c7f06a18c8d848/Tools/Create_Smooth_Font/Create_font/Create_font.pde#L396) the script was running properly, because it's formerly defined above.

**Comment by Bodmer** [comment, marked as answer]
I do not get the same results.

I did the following:
1. Load the Create_Font.pde sketch direct from the library TFT_eSPI\Tools\Create_Smooth_Font folder.
2. Ran with the default setting - all OK Final-Frontier20 font generated
3. Make a single edit to the file - change fontNumber to 409:
          int fontNumber = 409; 
   Ran again and this created font Yu Gothic UI Light20 as expected and without error

I delete the line you suggest and run again with fontNumber still 409 and the sketch crashes and has to be terminated

Not sure if you made any other edits that caused the problem, would be helpful if you try exactly the above with the original master sketch.

**Reply by u4mzu4** [reply]
Thanks for your quick answer! 
With Yu Gothic Light it's working properly for me as well, with Logisoso (which is int the System_Font_List.txt but installed later), the failure 
""Logisoso" is not available, so another font will be used. Use PFont.list() to show available fonts."
happened and font is overwritten with the default one. That's strange a little bit, becuse the file itself is next to the others and installed properly, I can choose in MS Word for example under Win10.
That was the reason why I moved the Logisoso.ttf file to the data folder, and modified the original code. Maybe it's just some access right issue, but the workaround is working, so no further activity needed I think.

**Reply by u4mzu4** [reply]
Only for the sake of posterity:
There is a Processing issue with some fonts, eg. TypeKit, which are available in the PFont.list(), however you can not load them, only with direct canonical path names.
Processing issue [5800](https://github.com/processing/processing/issues/5800).

**Reply by Bodmer** [reply]
Thanks for the info, that expains it!

**Comment by PaulskPt** [comment]
Hi, I recently successfully converted a .ttf font chosen and downloaded from Google Fonts using Processing. But it was a struggle. I experienced that Processing demands a certain folder and file structure. I took some screenshot of my folder and file setup, put the images together in one image and annotated it. See the attachment

![Folder_and_file_struct_needed_for_Processing_to_work_OK](https://user-images.githubusercontent.com/9107950/139557554-dbced06c-e329-43d0-bb59-f3e664b59418.png)
.
